test/core.jl

```diff
@@ -145,6 +145,16 @@ A = @inferred(AxisArray(reshape(1:24, 2,3,4),
               Axis{:y}(1//10:1//10:3//10),
               Axis{:z}(["a", "b", "c", "d"])))
 
+# recursive constructor
+@test A === @inferred AxisArray(A)
+@test axisnames(AxisArray(A, Axis{:yoyo}(1:length(A[Axis{:x}])))) == (:yoyo, :y, :z)
+@test typeof(AxisArray(A, Axis{:yoyo}(1:length(A[Axis{:x}]))).data) === typeof(A.data)
+@test axisnames(AxisArray(A, :something, :in, :the)) == (:something, :in, :the)
+@test typeof(AxisArray(A, :way, :you, :move).data) === typeof(A.data)
+@test axisnames(AxisArray(A, (:c, :a, :b), (2, 3, 4))) == (:c, :a, :b)
+@inferred AxisArray(A, Axis{:yoyo}(1:2))
```

Comment:
Maybe `Axis{:yoyo}(1:length(A[Axis{:x}]))`, just for consistency's sake with the above?